### ISSUE TITLE
Display predicted classes for a input image with Alexnet

### ISSUE THREAD

**Issue by deepali-c** [tflearn/tflearn#355]
I have trained Alexnet model with my dataset. The dataset has 6 classes. Next I want to use the model for predictions, so that it displays the top 3 classes predicted for a given input image. 

For this I am using model.predict() method. For a single input image the output of predict() is- 
[[0.38271743059158325, 6.913009656273061e-06, 0.01705058105289936, 0.5627931952476501, 0.002256882842630148, 0.0018134841229766607]

Reverse sort on the output gives the following:
[0.5627931952476501, 0.38271743059158325, 0.01705058105289936, 0.002256882842630148, 0.0018134841229766607, 6.913009656273061e-06]

The original indices corresponding to these are - [3,0,2,4,5,1].  Is there some way to map the class indices to their names? Please advise.

**Comment by aymericdamien** [comment]
You case ues numpy `argsort` function.

**Comment by deepali-c** [comment]
Perhaps my query is not so clear. Let me explain what I am trying to do: Suppose I have the following folders in the training data:
```
n01318894
n01324610
n01503061
n01844917
n01887787
n11669921
```
Such that when TF learn data_utils is used to populate the training data set then it maps the folders with the following labels:
```
Folder   Label
n01318894 0
n01324610 1
n01503061 2
n01844917 3
n01887787 4
n11669921 5
```
So in predict output I am getting the array which contains the labels for the predicted classes, but is there some direct way to get the corresponding folder name as well. 

Like in the example above the predict output is `[3,0,2,4,5,1].` So is there some way in TFLearn to map it to the corresponding folder name - `n01844917`.  Please advise.

**Comment by PavelGerashchenko** [comment]
Same issue. Result of `model.predict()` is shuffled.

**Comment by aymericdamien** [comment]
@deepali-c predict output probabilities of the neural network, so you have to do the mapping by yourself. If you have a dictionary, or a list of all labels, maybe simply using:
```
p_name = [folders_list[id] for id in p_ids]
```

**Comment by deepali-c** [comment]
@aymericdamien, thanks a bunch. I have done something similar by creating mapping in my code.

**Comment by maikelronnau** [comment]
Try `model.predict_label()`.
